extensions/amp-story/1.0/amp-story-desktop-one-panel.css

```diff
@@ -7,33 +7,32 @@
  * css/amp-story-player-shadow.css
  */
 
-/** Variables shared between page, system layer and pagination buttons. */
-.i-amphtml-story-desktop-one-panel {
-  --i-amphtml-story-desktop-one-panel-ratio: 69 / 116;
-  --i-amphtml-story-desktop-one-panel-responsive-margin: max(74px, 8.25vh);
-    /** Calculates panel height by subtracting responsive vertical margin. */
-  --i-amphtml-story-desktop-one-panel-height: calc(100vh - var(--i-amphtml-story-desktop-one-panel-responsive-margin) * 2);
-  --i-amphtml-story-desktop-one-panel-width: calc(var(--i-amphtml-story-desktop-one-panel-height) * var(--i-amphtml-story-desktop-one-panel-ratio));
-  --i-amphtml-story-desktop-one-panel-border-radius: 10px;
-}
-
-@media(max-height: 756px) {
-  .i-amphtml-story-desktop-one-panel {
-    --i-amphtml-story-desktop-one-panel-responsive-margin: 0;
-    --i-amphtml-story-desktop-one-panel-width: calc(100vh * var(--i-amphtml-story-desktop-one-panel-ratio));
-    --i-amphtml-story-desktop-one-panel-border-radius: 0;
+@media(min-aspect-ratio: 3/4) {
+  /** Variables shared between page, system layer and pagination buttons. */
+  :root:not([data-story-supports-landscape]) {
+    --i-amphtml-story-desktop-one-panel-ratio: 69 / 116;
+    --i-amphtml-story-desktop-one-panel-responsive-margin: max(74px, 8.25vh);
+      /** Calculates panel height by subtracting responsive vertical margin. */
+    --story-page-vh: calc(100vh - var(--i-amphtml-story-desktop-one-panel-responsive-margin) * 2);
+    --story-page-vw: calc(var(--i-amphtml-story-desktop-one-panel-height) * var(--i-amphtml-story-desktop-one-panel-ratio));
+    --i-amphtml-story-desktop-one-panel-border-radius: 10px;
   }
-}
 
-@media(max-height: 538px) {
-  .i-amphtml-story-desktop-one-panel {
-    --i-amphtml-story-desktop-one-panel-ratio: 3 / 4;
+  @media (max-height: 756px) {
+    :root:not([data-story-supports-landscape]) {
+      --i-amphtml-story-desktop-one-panel-responsive-margin: 0;
+      --i-amphtml-story-desktop-one-panel-width: calc(100vh * var(--i-amphtml-story-desktop-one-panel-ratio));
+      --i-amphtml-story-desktop-one-panel-border-radius: 0;
+    }
   }
-}
 
-.i-amphtml-story-desktop-one-panel amp-story-page {
-  width: var(--i-amphtml-story-desktop-one-panel-width) !important;
-  height: var(--i-amphtml-story-desktop-one-panel-height) !important;
-  border-radius: var(--i-amphtml-story-desktop-one-panel-border-radius) !important;
-  margin: auto !important;
+  @media(max-height: 538px) {
+    :root:not([data-story-supports-landscape]) {
+      --i-amphtml-story-desktop-one-panel-ratio: 3 / 4;
+    }
+  }
+  :root:not([data-story-supports-landscape]) amp-story-page {
+    border-radius: var(--i-amphtml-story-desktop-one-panel-border-radius) !important;
+    margin: auto !important;
+  }
```

Comment:
I think this is going in a very good direction but should be pushed further. The fact that `--story-page-vh/w` is in 2 CSS (not counting the JS that should be removed) places and they're overriding each other is problematic. Can you factorize further and make sure there's only one source of truth?

Reply:
Makes sense to remove the JS and rely on CSS. There will need to be 2 CSS locations where it's set though, one for fullbleed (mobile + landscape), and one for one-panel. They calculate the page size differently

Reply:
My question was about how to factorize that so all the values for `--story-page-vh/vw` are set from the same place. Ideally in software you don't override variables randomly in different files, but define a single source of truth

Reply:
Oh I misunderstood that part of the question. As for "in software you don't override variables randomly in different files", this is what happens in OOP when you have a subclass, that you can override/change the variables needed in the file of the subclass (in a way one-panel.css is a derivative of the fullbleed because it reuses the fullbleed styles and overrides some rules).

I'm thinking: what if we create a file `amp-story-units.css`? But if we have more UI types in the future it could get messier, and we'd have to move all the one-panel code to that file to keep it clean (potentially moving all UI types to that file).